mewarpx/mewarpx/diags_store/checkpoint_diagnostic.py

```diff
@@ -0,0 +1,45 @@
+"""Class for installing a checkpoint diagnostic"""
+
+from mewarpx.mwxrun import mwxrun
+from mewarpx.diags_store.diag_base import WarpXDiagnostic
+
+from pywarpx import picmi
+
+import logging
+
+logger = logging.getLogger(__name__)
+
+
+class CheckPointDiagnostic(WarpXDiagnostic):
+    def __init__(self, diag_steps, name, grid=None, write_dir=".", **kwargs):
+        """
+        This class is a wrapper for creating checkpoints from which
+        to restart a simulation from
+
+        Arguments:
+            diag_steps (int): Run the diagnostic with this period.
+                Also plot on this period if enabled.
+            grid (picmi grid): The grid to be passed into the picmi checkpoint diagnostic.
```

Comment:
Why is grid an optional parameter? It doesn't look like it is used anywhere.

Reply:
Ah yep, it's not necessary with the current Picmi checkpoint.